tests/common/permissions.rs

```diff
@@ -0,0 +1,992 @@
+    team_id: &str,
+    project_permissions: Option<ProjectPermissions>,
+    organization_permissions: Option<OrganizationPermissions>,
+    test_env: &TestEnvironment,
+) {
+    let api = &test_env.v2;
+
+    // Invite user
+    let resp = api
+        .add_user_to_team(
+            team_id,
+            user_id,
+            project_permissions,
+            organization_permissions,
+            ADMIN_USER_PAT,
+        )
+        .await;
+    assert!(resp.status().is_success());
+
+    // Accept invitation
+    let resp = api.join_team(team_id, user_pat).await;
+    assert!(resp.status().is_success());
+}
+
+async fn modify_user_team_permissions(
+    user_id: &str,
+    team_id: &str,
+    permissions: Option<ProjectPermissions>,
+    organization_permissions: Option<OrganizationPermissions>,
+    test_env: &TestEnvironment,
+) {
+    let api = &test_env.v2;
+
+    // Send invitation to user
+    let resp = api
+        .edit_team_member(
+            team_id,
+            user_id,
+            json!({
+                "permissions" : permissions.map(|p| p.bits()),
+                "organization_permissions" : organization_permissions.map(|p| p.bits()),
+            }),
+            ADMIN_USER_PAT,
+        )
+        .await;
+    assert!(resp.status().is_success());
+}
+
+async fn remove_user_from_team(user_id: &str, team_id: &str, test_env: &TestEnvironment) {
+    // Send invitation to user
+    let api = &test_env.v2;
+    let resp = api.remove_from_team(team_id, user_id, ADMIN_USER_PAT).await;
+    assert!(resp.status().is_success());
+}
+
+async fn get_project_permissions(
+    user_id: &str,
+    user_pat: &str,
+    project_id: &str,
+    test_env: &TestEnvironment,
+) -> ProjectPermissions {
+    let resp = test_env.v2.get_project_members(project_id, user_pat).await;
+    let permissions = if resp.status().as_u16() == 200 {
+        let value: serde_json::Value = test::read_body_json(resp).await;
+        value
+            .as_array()
+            .unwrap()
+            .iter()
+            .find(|member| member["user"]["id"].as_str().unwrap() == user_id)
+            .map(|member| member["permissions"].as_u64().unwrap())
+            .unwrap_or_default()
+    } else {
+        0
+    };
+
+    ProjectPermissions::from_bits_truncate(permissions)
+}
+
+async fn get_organization_permissions(
+    user_id: &str,
+    user_pat: &str,
+    organization_id: &str,
+    test_env: &TestEnvironment,
+) -> OrganizationPermissions {
+    let api = &test_env.v2;
+    let resp = api
+        .get_organization_members(organization_id, user_pat)
+        .await;
+    let permissions = if resp.status().as_u16() == 200 {
+        let value: serde_json::Value = test::read_body_json(resp).await;
+        value
+            .as_array()
+            .unwrap()
+            .iter()
+            .find(|member| member["user"]["id"].as_str().unwrap() == user_id)
+            .map(|member| member["organization_permissions"].as_u64().unwrap())
```

Comment:
for these ones I think it would be better to put the permissions test of each object in it's own respective file. i think it would be more organized and makes more sense to me